app/models/proposal.rb

```diff
@@ -206,6 +206,26 @@ def partial_approve
     end
   end
 
+  # Is the user getting emails for this proposal
+  def is_user_tracking? user
+    users = [self.requester]
+    users += self.approvers + self.observers
+    users.include? user
+  end
+
+  def subscribe user
+    if !self.is_user_tracking? user
```

Comment:
Do we need this if the emails are getting deduplicated later? If not, less logic to carry around.

Note that you'll want to merge in master for https://github.com/18F/C2/blob/6ce10e31faae61e5a428763ccaa3d9f48b1c4827/app/models/proposal.rb#L146